### ISSUE TITLE
SPFx solution with Graph api not working as Teams Tab (in native client)

### ISSUE THREAD

**Issue by MaheshakaMudli** [SharePoint/sp-dev-docs#7034]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

other (enter in the "Additional environment details" area below)

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [X] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

- Teams desktop app (native client)
- SPFx 1.11.0
- Node.js 10.24.1

### Describe the bug / error

SPFx webpart with Graph API hosted as teams tab is working fine in browsers but failing in native client app and SharePoint app too. We have started facing this issue recently as it was working before. On one of our dev tenant it is working fine but not working in prod environment. 

### Steps to reproduce

1. Add a sharepoint page as Teams tab with SPFX webpart and graph api in it.
2. Open the tab in native client.
3. The graph api gives 403 Forbidden error.
4. On checking the network the url which throws this error is 
  **https://tenant.sharepoint.com/_api/Microsoft.Sharepoint.Internal.ClientSideComponent.Token.AcquireOBOToken?resource=%27ece7c1fb-fcef-4195-ae0c-65fe47a4602e%27&clientId=%27e427a6c3-f470-4c33-9af7-6b0e1fd21eda%27**

### Expected behavior

The teams tab with graph api should load the data in native client too.

**Comment by VesaJuvonen** [comment]
@MaheshakaMudli - We had recently a fix rolling for this which should resolve the issue in both web and desktop client. Is this still an issue in your environment? Getting the updated status, so that we can follow up as needed. Thanks for confirming the current situation.

**Comment by patmill** [comment]
Sorry - quick update.  The fix is rolling, but is not yet 100%.

**Comment by xstos** [comment]
We have a very similar issue in our corporate spfx solution. Works in browser-based teams, but not in native client. We have a different environment that works fine. We opened the developer tools console in teams native and saw this in the network tab:

https://<redacted>.sharepoint.com/sites/Finance/_api/Microsoft.SharePoint.Internal.ClientSideComponent.Token.AcquireOBOToken?resource=%27api://<redacted>:23340/<redacted>%27&clientId=%27<redacted>%27
Request Method:GET
Status Code:400 Bad Request

Shall we open a separate ticket? 

Also, it's frustrating that these errors are swallowed in teams with no indication of what's going on. Clicking to launch our addon does nothing. Perhaps teams needs a toast notification or something to indicate errors.

**Comment by jonathanhotono** [comment]
Hi @VesaJuvonen , this issue also happened on mobile teams client where spfx cannot call graph api and it is quite difficult to debug the network request/code through teams tab mobile on both android/ios. This issue happened across multiple tenants.

**Comment by harshdamaniahd** [comment]
for spfx 1.10 , any solution deployed to teams has stopped working

**Comment by AJIXuMuK** [comment]
Hey @harshdamaniahd - seems like your issues is being discussed in #7311

**Comment by harshdamaniahd** [comment]
> Hey @harshdamaniahd - seems like your issues is being discussed in #7311 

is it a known issue?

**Comment by AJIXuMuK** [comment]
@harshdamaniahd yes. And it seems like you've already commented on the related Teams issue [here](https://github.com/MicrosoftDocs/msteams-docs/issues/4110)

**Comment by lucabandMSFT** [comment]
Is the issue still happening?

**Comment by sortegamartin** [comment]
It seems to be that yes, in my tenant it is happening again

**Comment by lucabandMSFT** [comment]
hey @sortegamartin: Yes, we found an issue for that specific scenario and we are working on a fix right now.. we are planning to be able to release that by mid/end of November 100% in production. I will update this issue as soon as we have it deployed.

**Comment by PhilGude** [comment]
@lucabandMSFT So we received the update. But in my case the particular issue still here.

**Comment by lucabandMSFT** [comment]
@PhilGude , thanks. Can you please navigate to the API Access page in SharePoint Online tenant admin with an account that has Global admin role and see if that solves the issue?

**Comment by PhilGude** [comment]
@lucabandMSFT Thanks for the fast feedback.  Visited the API Access page again with the Global tenant admin role and it did change the Application ID URI in Azure's SharePoint Online Client Extensibility Web Application. And I can now confirm it works both on the Teams Fat client and the SharePoint mobile app. Thanks a lot !

**Comment by lucabandMSFT** [comment]
@PhilGude , that is GREAT NEWS! Tanks to confirming it.
